### ISSUE TITLE
Support `JRubyExec` in Java and Kotlin

### ISSUE THREAD

**Issue by gmazzo** [jruby-gradle/jruby-gradle-plugin#397]
Right now is not possible to extend `JRubyExec` unless you write your extension class in `groovy`.
You'll get:
```
Class 'XXX' is not abstract and does not implement abstract base class member public abstract fun getProperty(p0: String!): Any! defined in com.github.jrubygradle.JRubyExec
```

I think you should move the source code of those tasks to Java or Kotlin to support the 3 languages.

**Comment by ysb33r** [comment]
Are you using 2.0.0-alpha.X ?

**Comment by gmazzo** [comment]
I've tried alpha.3, yes, and got the same result.

**Comment by ysb33r** [comment]
When I get a chance I'll see if I can write a test for it.

**Comment by ysb33r** [comment]
I assume you tried to extend using Kotlin?

**Comment by gmazzo** [comment]
Do you really need groovy classes? Just regular Java classes will work for any platform. Actually, it would be nice to add a test with KTS too, as Kotlin scripts are getting relevance vs Groovy ones and the is a lot a plugins not designed for it. You should not rely on Closure for instance and replace it with Action, from Gradle's API

**Comment by ysb33r** [comment]
Yes, it is a lot more work to write wide-Gradle compatibility in Java.

You are right about Closures. If you find places in the 2.x API where Action-based methods are missing, please raise a defect so that support can be added.
